numba/npyufunc/tbbpool.cpp

```diff
@@ -34,14 +34,93 @@ add_task(void *fn, void *args, void *dims, void *steps, void *data) {
     });
 }
 
+#define _DEBUG 0
+
+static void
+parallel_for_1d(void *fn, char **args, size_t *dimensions, size_t *steps, void *data,
+                size_t inner_ndim, size_t array_count, size_t)
+{
+    static bool printed = false;
+    if(!printed) {
+        puts("Using parallel_for_1d");
+        printed = true;
+    }
+
+    //     args = <ir.Argument '.1' of type i8**>,
+    //     dimensions = <ir.Argument '.2' of type i64*>
+    //     steps = <ir.Argument '.3' of type i64*>
+    //     data = <ir.Argument '.4' of type i8*>
+
+    const size_t arg_len = (inner_ndim + 1);
+
+    if(_DEBUG)
+    {
+        printf("inner_ndim: %ld\n",inner_ndim);
+        printf("arg_len: %ld\n", arg_len);
+        printf("total: %ld\n", dimensions[0]);
+        printf("dimensions: ");
+        for(size_t j = 0; j < arg_len; j++)
+            printf("%ld, ", ((size_t *)dimensions)[j]);
+        printf("\nsteps: ");
+        for(size_t j = 0; j < array_count; j++)
+            printf("%ld, ", steps[j]);
+        printf("\n*args: ");
+        for(size_t j = 0; j < array_count; j++)
+            printf("%p, ", (void *)args[j]);
+        printf("\n");
+    }
+
+    using range_t = tbb::blocked_range<size_t>;
+    tbb::parallel_for(range_t(0, dimensions[0]), [=](const range_t &range) {
+        size_t * count_space = (size_t *)alloca(sizeof(size_t) * arg_len);
+        char ** array_arg_space = (char**)alloca(sizeof(char*) * array_count);
+        memcpy(count_space, dimensions, arg_len * sizeof(size_t));
+        count_space[0] = range.size();
+
+        if(_DEBUG)
+        {
+            printf("THREAD %p:", count_space);
+            printf("count_space: ");
+            for(size_t j = 0; j < arg_len; j++)
+                printf("%ld, ", count_space[j]);
+            printf("\n");
+        }
+        for(size_t j = 0; j < array_count; j++)
+        {
+            char * base = args[j];
+            size_t step = steps[j];
+            ptrdiff_t offset = step * range.begin();
+            array_arg_space[j] = (char *)(base + offset);
```

Comment:
@stuartarchibald are you sure that it goes for all the arguments this way?  step is taken for each array or for each dimension? because I believe we want steps[0] only.

Reply:
IIRC it's steps for each array. https://github.com/anton-malakhov/numba/blob/575962557f6b681460d31d0d736bb901c94c2cb9/numba/npyufunc/parallel.py#L143-L147